lib/software_challenge_client/game_state.rb

```diff
@@ -21,12 +21,13 @@ class GameState
   # @!attribute [rw] startColor
   # @return [Color] Die Farbe, die zuerst legen darf
   attr_accessor :start_color
-  # @!attribute [rw] current_color_index
-  # @return [Color] Der jetzige Index in der Zug Reihenfolge der Farben.
-  attr_accessor :current_color_index
-  # @!attribute [rw] ordered_colors
+   # @!attribute [rw] valid_colors
   # @return [Array<Color>] Ein Array aller Farben die ziehen können in 
   #                        der Reihenfolge in der sie drankommen
```

Comment:
This isn't quite correct, `valid_colors` is serialised as set, without order. Since this set can potentially change each turn, it doesn't make sense to put them in order (for order, they can just look at `ordered_colors`